pages/learn/deploy/deployment.md

```diff
@@ -56,24 +56,24 @@ The emulated builds will also happen on the rare occasion that the native ARM bu
 
 ![How {{$names.company.lower}} deploy works](/img/common/deployment/balena-deploy.jpg)
 
-`{{$names.company.lower}} deploy` will build all your container images on the machine the command is run on (or on a specified docker daemon), and upon success, it will upload the images to the {{$names.cloud.lower}} image registry and then create a release entry in the [{{$names.company.lower}} API][api] database. The devices in the application will then be notified of a new release and download it. In order to build containers you will need to have [Docker installed][docker-install] on your development machine, and you should be able to execute [Docker commands as a non-root user][docker-post-install]. It is not necessary to install Docker on your development machine if you choose to use a device running {{$names.os.lower}} to build the images (a [development image][development-image] is then required), by specifying a docker daemon's IP address and port number with the relevant command-line options.
+With  `{{$names.company.lower}} build` container images are built on your development machine or on a remote machine, by specifying a docker daemon's IP address and port number with the relevant command-line options (for example a device running a {{$names.os.lower}} [development image][development-image]). Depending on your fleet's targeted CPU architecture builds will be run emulated via [qemu][qemu].
+
+If you are building your own container images, `{{$names.company.lower}} deploy` will upload the images to the {{$names.cloud.lower}} image registry and then create a release entry in the [{{$names.company.lower}} API][api] database. The devices in the application will then be notified of a new release and download it. Should `{{$names.company.lower}} deploy` not find the required images on the specified docker daemon it will automatically trigger a build.
 
 Like `{{$names.company.lower}} push` it is also independent of git, and you can use any version control system you wish. It is also possible to make use of [private base images](#private-base-images).
 
 __Note:__ Currently `{{$names.company.lower}} deploy` does not support the [build time secrets](#build-time-secrets-and-variables) feature.
 
-It's also possible to use the `{{$names.company.lower}} build` command without actually deploying. This command has all the same functionality as `{{$names.company.lower}} deploy`, but it does not upload the images to the registry or create a release on the API. This command can be useful if you want your CI/CD system to first run built images through some testing and validation stage before finally doing the deploy.
-
 ### Additional Options
```

Comment:
I would actually prefer to remove the incomplete documentation of those command options and just have a link to the relevant cli reference section... it's just duplicated effort this way.